packages/iov-bns/src/bnsconnection.ts

```diff
@@ -623,18 +606,25 @@ export class BnsConnection implements BcpAtomicSwapConnection {
   }
 
   public async getFeeQuote(transaction: UnsignedTransaction): Promise<Fee> {
-    if (isBnsTx(transaction)) {
-      const firstToken = (await this.getAllTickers())[0];
-      return {
-        tokens: {
-          quantity: "0",
-          fractionalDigits: firstToken.fractionalDigits,
-          tokenTicker: firstToken.tokenTicker,
-        },
-      };
+    if (!isBnsTx(transaction)) {
+      throw new Error("Received transaction of unsupported kind.");
+    }
+    // get gconf fee
+    // TODO: get product fee for this type
+    // take maximum
+    const res = await this.query("/", Encoding.toAscii("gconf:cash:minimal_fee"));
+    if (res.results.length === 0) {
+      throw new Error("No fee defined");
+    } else if (res.results.length > 1) {
+      throw new Error("Received unexpected number of fees");
```

Comment:
This can be more compact: `res.results.length !=== 1`, which works for 0 as well

Reply:
:+1: